pkg/restore/pipeline_items.go

```diff
@@ -0,0 +1,310 @@
+// Copyright 2020 PingCAP, Inc. Licensed under Apache-2.0.
+
+package restore
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/kvproto/pkg/backup"
+	"github.com/pingcap/log"
+	"github.com/pingcap/parser/model"
+	"go.uber.org/zap"
+
+	"github.com/pingcap/br/pkg/conn"
+	"github.com/pingcap/br/pkg/glue"
+	"github.com/pingcap/br/pkg/rtree"
+	"github.com/pingcap/br/pkg/utils"
+)
+
+const (
+	defaultBatcherOutputChannelSize = 1024
+)
+
+// CreatedTable is a table is created on restore process,
+// but not yet filled by data.
+type CreatedTable struct {
+	RewriteRule *RewriteRules
+	Table       *model.TableInfo
+	OldTable    *utils.Table
+}
+
+// TableWithRange is a CreatedTable that has been bind to some of key ranges.
+type TableWithRange struct {
+	CreatedTable
+
+	Range []rtree.Range
+}
+
+// Batcher collects ranges to restore and send batching split/ingest request.
+type Batcher struct {
+	cachedTables   []TableWithRange
+	cachedTablesMu *sync.Mutex
+	rewriteRules   *RewriteRules
```

Comment:
explain the relationship between this field and the `RewriteRule` fields in `cachedTables`?

Reply:
This is a 'collection' of all table's `RewriteRule`.

I think leave it here instead of put it in sender(`Sender` can get those rules per table from `TableWithRange`), and should be refactored(as a matter of fact, I forgot remove this when extracting `Sender` from `Batcher`).